src/lifetracker/calendar/CalendarEntry.java

```diff
@@ -30,6 +30,14 @@
 
     EntryType getType();
 
+    void setRecurring(boolean recurring);
```

Comment:
I don't really think you need this method for the interface right? Usually when you set it to recurring you need a period. But if anyone could set the flag from the outside without providing a period, you might have problems later when you query the next date.